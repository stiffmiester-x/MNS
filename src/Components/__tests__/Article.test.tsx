import {render,screen , cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import PostsWithAxios from '../Article'

afterEach(()=>{
    cleanup();
})

test('to render the component',()=>{
    render(<PostsWithAxios/>)
    const element=screen.getByTestId('imagediv')
    expect(element).toBeInTheDocument();
    expect(element).toHaveClass('axioscontainer')

})

test('should check the text', () => {
    render(<PostsWithAxios/>)
    const element=screen.getByTestId('imagediv')
    expect(element).not.toContainHTML('<a>')

})

test('snapshots', () => {
    const tree=renderer.create(<PostsWithAxios/>).toJSON()
    console.log(tree)
    expect(tree).toMatchSnapshot()
    
})

