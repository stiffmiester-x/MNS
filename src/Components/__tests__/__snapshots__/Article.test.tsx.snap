// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots 1`] = `
Array [
  <img
    alt="logo"
    className="logo"
    src="Styleandliving_Icon.webp"
  />,
  <div
    className="axioscontainer"
    data-testid="imagediv"
    onClick={[Function]}
  />,
]
`;
